crates/minecraft_packets/src/play/data/chunk_data.rs

```diff
@@ -94,7 +100,146 @@ impl ChunkData {
+                                            .into_iter()
+                                            .map(|msg| {
+                                                let text = msg.get_string().unwrap_or_default();
+
+                                                let processed_text = if protocol_version
+                                                    .is_before_inclusive(ProtocolVersion::V1_21_4)
+                                                {
+                                                    // For 1.21.4 and below, add quotes if not present
+                                                    if text.starts_with('"') && text.ends_with('"')
+                                                    {
+                                                        text.to_string()
+                                                    } else {
+                                                        format!("\"{}\"", text)
+                                                    }
+                                                } else {
+                                                    // For 1.21.5 and above, remove quotes if present
+                                                    text.strip_prefix('"')
+                                                        .and_then(|s| s.strip_suffix('"'))
+                                                        .map(String::from)
+                                                        .unwrap_or_else(|| text.to_string())
+                                                };
+
+                                                // Create a new NBT string with processed value
+                                                Nbt::String {
+                                                    name: msg.get_name(),
+                                                    value: processed_text,
+                                                }
+                                            })
+                                            .collect();
+
+                                        // Rebuild x2
+                                        Nbt::List {
+                                            name: msg_name.clone(),
+                                            value: edited_messages,
+                                            tag_type: 8,
+                                        }
+                                    }
+                                    other => other,
+                                })
+                                .collect();
+
+                            // Rebuild x3
+                            Nbt::Compound {
+                                name: text_name.clone(),
+                                value: new_text,
+                            }
+                        }
+                        other => other,
+                    })
+                    .collect();
+
+                // Rebuild x4
+                Nbt::Compound {
+                    name: name.clone(),
```

Comment:
No need to clone here

```suggestion
                    name,
```